web/ts/api/admin-lecture-list.ts

```diff
@@ -0,0 +1,296 @@
+    startDate: Date;
+    startDateFormatted: string;
+    startTimeFormatted: string;
+    endDate: Date;
+    endDateFormatted: string;
+    endTimeFormatted: string;
+
+    // Clientside pseudo fields
+    newCombinedVideo: File | null;
+    newPresentationVideo: File | null;
+    newCameraVideo: File | null;
+}
+
+export interface DownloadableVOD {
+    FriendlyName: string;
+    DownloadURL: string;
+}
+
+/**
+ * REST API Wrapper for /api/stream/:id/sections
+ */
+export const AdminLectureList = {
+    /**
+     * Fetches all lectures for a course
+     * @param courseId
+     */
+    get: async function (courseId: number): Promise<Lecture[]> {
+        const result = await get(`/api/course/${courseId}/lectures`);
+        return result["streams"];
+    },
+
+    /**
+     * Adds a new lecture to a course.
+     * @param courseId
+     * @param request
+     */
+    add: async function (courseId: number, request: object) {
+        return post(`/api/stream/${courseId}/sections`, request);
+    },
+
+    /**
+     * Updates metadata of a lecture.
+     * @param courseId
+     * @param lectureId
+     * @param request
+     */
+    updateMetadata: async function (courseId: number, lectureId: number, request: UpdateLectureMetaRequest) {
+        console.log({
+            request,
+        });
+        const promises = [];
+        if (request.name !== undefined) {
+            promises.push(
+                postData(`/api/course/${courseId}/renameLecture/${lectureId}`, {
+                    name: request.name,
+                }),
+            );
+        }
+
+        if (request.description !== undefined) {
+            promises.push(
+                putData(`/api/course/${courseId}/updateDescription/${lectureId}`, {
+                    name: request.description,
+                }),
+            );
+        }
+
+        if (request.lectureHallId !== undefined) {
+            promises.push(
+                postData("/api/setLectureHall", {
+                    streamIds: [lectureId],
+                    lectureHall: request.lectureHallId,
+                }),
+            );
+        }
+
+        if (request.isChatEnabled !== undefined) {
+            promises.push(
+                patchData(`/api/stream/${lectureId}/chat/enabled`, {
+                    lectureId,
+                    isChatEnabled: request.isChatEnabled,
+                }),
+            );
+        }
+
+        const errors = (await Promise.all(promises)).filter((res) => res.status !== StatusCodes.OK);
+        if (errors.length > 0) {
+            console.error(errors);
+            throw Error("Failed to update all data.");
+        }
+    },
+
+    /**
+     * Distributes the lecture metadata of given lecture to all lectures in its series.
+     * @param courseId
+     * @param lectureId
+     */
+    saveSeriesMetadata: async (courseId: number, lectureId: number) => {
+        return await postData(`/api/course/${courseId}/updateLectureSeries/${lectureId}`);
```

Comment:
No actually not, I missed that `../utilities/fetch-wrappers` exists. Thanks for the feedback makes the code even shorter because error handling is already in `../utilities/fetch-wrappers` 🚀